packages/core/src/codecatalyst/auth.ts

```diff
@@ -81,8 +105,21 @@ export class CodeCatalystAuthenticationProvider {
         return this.secondaryAuth.hasSavedConnection
     }
 
+    public setScopeExpired(isExpired: boolean) {
+        if (!this.activeConnection) return
```

Comment:
This is hard to follow. `setScopeExpired` just randomly will do nothing if there isn't an active connection? Can the caller make this decision instead?